wcfsetup/install/files/js/WCF.Attachment.js

```diff
@@ -308,6 +315,85 @@ if (COMPILER_TARGET_DEFAULT) {
 			// remove failed uploads
 			this._fileListSelector.children('li.uploadFailed').remove();
 			
+			if (this._options.enableAutoScale) {
+				var self = this;
+				var maxSize = this._buttonSelector.data('maxSize');
+				
+				var promise = new Promise(function (resolve) {
+					require(['ImageResizer'], function (ImageResizer) {
+						var promises = [];
+						
+						// Create an ImageResizer instance
+						var resizer = new ImageResizer()
+							.setMaxWidth(self._options.autoScaleMaxWidth)
+							.setMaxHeight(self._options.autoScaleMaxHeight)
+							.setFileType(self._options.autoScaleFileType)
+							.setQuality(self._options.autoScaleQuality);
+						
+						for (var i = 0; i < files.length; i++) {
+							if (files[i].type.match(/^image\//i)) {
+								(function (i) {
+									promises.push(new Promise(function (resolve) {
+										var file = files[i];
+										
+										if (file.blob) {
+											var name = file.name;
+											file = file.blob;
+											file.name = name;
+										}
+										
+										var reader = new FileReader();
+										var image = new Image();
+										var needsResize = file.size > maxSize;
+										
+										reader.onloadend = function () {
+											image.src = reader.result;
+										};
+										
+										image.onload = function () {
+											needsResize = needsResize
+												    || image.width > self._options.autoScaleMaxWidth
+												    || image.height > self._options.autoScaleMaxHeight;
+											
+											if (!needsResize) {
+												resolve(file);
+											}
+											else {
+												resizer.resize(image)
+													.then(function () {
+														return resizer.getFile(file.name)
+													})
+													.then(resolve)
+													.catch(function (error) {
```

Comment:
This catch should be moved before `.then(resolve)` and simply `return file`. Otherwise you are going to catch errors in the invocation of `resolve`, leading calling `resolve` twice which violates the Promise's contract.